pilot/pkg/networking/plugin/authn/authentication.go

```diff
@@ -128,16 +131,42 @@ func ConvertPolicyToJwtConfig(policy *authn.Policy) *jwtfilter.JwtAuthentication
 		AllowMissingOrFailed: true,
 	}
 	for _, policyJwt := range policyJwts {
-		hostname, port, _, err := model.ParseJwksURI(policyJwt.JwksUri)
-		if err != nil {
-			log.Errorf("Cannot parse jwks_uri %q: %v", policyJwt.JwksUri, err)
-			continue
+		jwt := &jwtfilter.JwtRule{
+			Issuer:               policyJwt.Issuer,
+			Audiences:            policyJwt.Audiences,
+			ForwardPayloadHeader: OutputLocationForJwtIssuer(policyJwt.Issuer),
+			Forward:              true,
 		}
 
-		jwt := &jwtfilter.JwtRule{
-			Issuer:    policyJwt.Issuer,
-			Audiences: policyJwt.Audiences,
-			JwksSourceSpecifier: &jwtfilter.JwtRule_RemoteJwks{
+		for _, location := range policyJwt.JwtHeaders {
+			jwt.FromHeaders = append(jwt.FromHeaders, &jwtfilter.JwtHeader{
+				Name: location,
+			})
+		}
+		jwt.FromParams = policyJwt.JwtParams
+
+		if fetchPubKey {
+			jwtPubKey, err := model.Jwtkeyresolver.ResolveJwtPubKey(policyJwt.JwksUri)
+			if err != nil {
+				log.Warnf("Failed to fetch jwt public key from %q", policyJwt.JwksUri)
+			}
+
+			// Put empty string in config even if above ResolveJwtPubKey fails.
+			jwt.JwksSourceSpecifier = &jwtfilter.JwtRule_LocalJwks{
+				LocalJwks: &core.DataSource{
+					Specifier: &core.DataSource_InlineString{
+						InlineString: jwtPubKey,
+					},
+				},
+			}
+		} else {
+			hostname, port, _, err := model.ParseJwksURI(policyJwt.JwksUri)
+			if err != nil {
+				log.Errorf("Cannot parse jwks_uri %q: %v", policyJwt.JwksUri, err)
+				continue
```

Comment:
May be we should do the same as above in error case, as jwksURI is no longer guaranteed to be valid one. In fact, we may even remove the check.

Reply:
done(keep the check just for logging).